packages/codex-harness/scripts/codex_run.py

```
@@ -0,0 +1,393 @@
+    try:
+        argv = shlex.split(command)
+    except ValueError as exc:
+        passed = False
+        output = f"invalid command syntax: {exc}"
+        status = "failed"
+        result = {"command": command, "status": status, "summary": output[:2000]}
+        log_block = f"=== [{status.upper()}] {command} ===\n{output}"
+        return result, log_block
+
+    try:
+        completed = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)